test/integration/scheduler_perf/scheduler_bench_test.go

```diff
@@ -79,6 +85,88 @@ func BenchmarkSchedulingPodAntiAffinity(b *testing.B) {
+		{nodes: 5000, existingPods: 1000, minPods: 1000},
+	}
+	// The setup strategy creates pods with no volumes.
+	setupStrategy := testutils.NewSimpleWithControllerCreatePodStrategy("setup")
+	// The test strategy creates pods with a secret.
+	testBasePod := makeBasePodWithSecret()
+	testStrategy := testutils.NewCustomCreatePodStrategy(testBasePod)
+	for _, test := range tests {
+		name := fmt.Sprintf("%vNodes/%vPods", test.nodes, test.existingPods)
+		b.Run(name, func(b *testing.B) {
+			benchmarkScheduling(test.nodes, test.existingPods, test.minPods, defaultNodeStrategy, setupStrategy, testStrategy, b)
+		})
+	}
+}
+
+// BenchmarkSchedulingInTreePVs benchmarks the scheduling rate of pods with
+// in-tree volumes (used via PV/PVC). Nodes have default hardcoded attach limits
+// (39 for AWS EBS).
+func BenchmarkSchedulingInTreePVs(b *testing.B) {
+	tests := []struct{ nodes, existingPods, minPods int }{
+		{nodes: 500, existingPods: 250, minPods: 250},
+		{nodes: 500, existingPods: 5000, minPods: 250},
+		{nodes: 1000, existingPods: 1000, minPods: 500},
+		{nodes: 5000, existingPods: 1000, minPods: 1000},
+	}
+	// The setup strategy creates pods with no volumes.
+	setupStrategy := testutils.NewSimpleWithControllerCreatePodStrategy("setup")
+
+	// The test strategy creates pods with AWS EBS volume used via PV.
+	baseClaim := makeBasePersistentVolumeClaim()
+	basePod := makeBasePod()
+	testStrategy := testutils.NewCreatePodWithPersistentVolumeStrategy(baseClaim, awsVolumeFactory, basePod)
+	for _, test := range tests {
+		name := fmt.Sprintf("%vNodes/%vPods", test.nodes, test.existingPods)
+		b.Run(name, func(b *testing.B) {
+			benchmarkScheduling(test.nodes, test.existingPods, test.minPods, defaultNodeStrategy, setupStrategy, testStrategy, b)
+		})
+	}
+}
+
+// BenchmarkSchedulingInTreePVs benchmarks the scheduling rate of pods with
+// in-tree volumes (used via PV/PVC). Nodes have attach limits filled in
+// node.status.allocatable.
+func BenchmarkSchedulingCSIPVs(b *testing.B) {
+	tests := []struct{ nodes, existingPods, minPods int }{
+		{nodes: 500, existingPods: 250, minPods: 250},
+		{nodes: 500, existingPods: 5000, minPods: 250},
+		{nodes: 1000, existingPods: 1000, minPods: 500},
+		{nodes: 5000, existingPods: 1000, minPods: 1000},
+	}
+	// The setup strategy creates pods with no volumes.
+	setupStrategy := testutils.NewSimpleWithControllerCreatePodStrategy("setup")
+
+	// The test strategy creates pods with CSI volume via PV.
+	baseClaim := makeBasePersistentVolumeClaim()
+	basePod := makeBasePod()
+	testStrategy := testutils.NewCreatePodWithPersistentVolumeStrategy(baseClaim, csiVolumeFactory, basePod)
+
+	// Each node can use the same amount of CSI volumes as in-tree AWS volume
+	// plugin, so the results should be comparable with BenchmarkSchedulingInTreePVs.
+	driverKey := util.GetCSIAttachLimitKey(testCSIDriver)
+	allocatable := map[v1.ResourceName]string{
+		v1.ResourceName(driverKey): fmt.Sprintf("%d", util.DefaultMaxEBSVolumes),
```

Comment:
Can we make the limit used in the test configurable?

Reply:
IMO it's quite useless. In the most 'dense' test, each node gets 10 pods.

Reply:
Interesting why we don't try to max out the 100 pods/node limit in the benchmark.  I would prefer if we can test our "worst case" limits, ie ~100 volumes per node.